src/library-authoring/backup-restore/LibraryBackupPage.tsx

```diff
@@ -0,0 +1,211 @@
+import { LibraryBackupStatus } from '@src/library-authoring/backup-restore/data/constants';
+import { useCreateLibraryBackup, useGetLibraryBackupStatus } from '@src/library-authoring/backup-restore/data/hooks';
+import NotFoundAlert from '@src/generic/NotFoundAlert';
+import Header from '@src/header';
+import { useLibraryContext } from '@src/library-authoring/common/context/LibraryContext';
+import { useContentLibrary } from '@src/library-authoring/data/apiHooks';
+import messages from '@src/library-authoring/backup-restore/messages';
+
+export const LibraryBackupPage = () => {
+  const intl = useIntl();
+  const { libraryId } = useLibraryContext();
+  const [taskId, setTaskId] = useState<string>('');
+  const [isMutationInProgress, setIsMutationInProgress] = useState<boolean>(false);
+  const timeoutRef = useRef<NodeJS.Timeout | null>(null);
+  const { data: libraryData } = useContentLibrary(libraryId);
+
+  const mutation = useCreateLibraryBackup(libraryId);
+  const backupStatus = useGetLibraryBackupStatus(libraryId, taskId);
+
+  // Clean up timeout on unmount
+  useEffect(() => () => {
+    if (timeoutRef.current) {
+      clearTimeout(timeoutRef.current);
+    }
+  }, []);
+
+  const handleDownload = useCallback((url: string) => {
+    try {
+      // Create a temporary anchor element for better download handling
+      const link = document.createElement('a');
+      link.href = url;
+      link.download = `${libraryData?.slug || 'library'}-backup.tar.gz`;
+      document.body.appendChild(link);
+      link.click();
+      document.body.removeChild(link);
+    } catch (error) {
+      // Fallback to window.location.href if the above fails
+      window.location.href = url;
+    }
+  }, [libraryData?.slug]);
+
+  const handleDownloadBackup = useCallback(() => {
+    // If backup is ready, download it immediately
+    if (backupStatus.data?.state === LibraryBackupStatus.Succeeded && backupStatus.data.url) {
+      const fullUrl = `${getConfig().STUDIO_BASE_URL}${backupStatus.data.url}`;
+      handleDownload(fullUrl);
+      return;
+    }
+
+    // If no backup in progress, create a new one
+    if (!taskId) {
+      setIsMutationInProgress(true);
+      mutation.mutate(undefined, {
+        onSuccess: (data) => {
+          setTaskId(data.task_id);
+          // Clear task id after 1 minutes to allow new backups
+          timeoutRef.current = setTimeout(() => {
+            setTaskId('');
+            setIsMutationInProgress(false);
+            timeoutRef.current = null;
+          }, 60 * 1000);
+        },
+        onError: () => {
+          setIsMutationInProgress(false);
+        },
+      });
+    }
+  }, [taskId, backupStatus.data, mutation, handleDownload]);
+
+  // Auto-download when backup becomes ready
+  useEffect(() => {
+    if (backupStatus.data?.state === LibraryBackupStatus.Succeeded && backupStatus.data.url) {
+      const fullUrl = `${getConfig().STUDIO_BASE_URL}${backupStatus.data.url}`;
```

Comment:
Is this `data.url` coming from an API? REST APIs should always be returning full URLs - I think we should update the API so we don't have to add `${getConfig().STUDIO_BASE_URL}` here in the frontend.


(can be follow-up PR)